sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -0,0 +1,332 @@
+ * @param[in] mqtt_user_name An empty #az_span with sufficient capacity to hold the MQTT user name.
+ * @param[out] out_mqtt_user_name The output #az_span containing the MQTT user name.
+ * @return #az_result.
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_user_name_get(
+    az_iot_pnp_client const* client,
+    az_span mqtt_user_name,
+    az_span* out_mqtt_user_name);
+
+/**
+ * @brief Gets the MQTT client id.
+ *
+ * The client id will be of the following format:
+ * {device_id}
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] mqtt_client_id An empty #az_span with sufficient capacity to hold the MQTT client id.
+ * @param[out] out_mqtt_client_id The output #az_span containing the MQTT client id.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_id_get(
+    az_iot_pnp_client const* client,
+    az_span mqtt_client_id,
+    az_span* out_mqtt_client_id);
+
+/**
+ *
+ * SAS Token APIs
+ *
+ *   Use the following APIs when the Shared Access Key is available to the application or stored
+ *   within a Hardware Security Module. The APIs are not necessary if X509 Client Certificate
+ *   Authentication is used.
+ */
+
+/**
+ * @brief Gets the Shared Access clear-text signature.
+ * @details The application must obtain a valid clear-text signature using this API, sign it using
+ *          HMAC-SHA256 using the Shared Access Key as password then Base64 encode the result.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] token_expiration_epoch_time The time, in seconds, from 1/1/1970.
+ * @param[in] signature An empty #az_span with sufficient capacity to hold the SAS signature.
+ * @param[out] out_signature The output #az_span containing the SAS signature.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_sas_signature_get(
+    az_iot_pnp_client const* client,
+    uint32_t token_expiration_epoch_time,
+    az_span signature,
+    az_span* out_signature);
+
+/**
+ * @brief Gets the MQTT password.
+ * @note The MQTT password must be an empty string if X509 Client certificates are used. Use this
+ *       API only when authenticating with SAS tokens.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] base64_hmac_sha256_signature The Base64 encoded value of the HMAC-SHA256(signature,
+ *                                         SharedAccessKey). The signature is obtained by using
+ *                                         #az_iot_pnp_client_sas_signature_get.
+ * @param[in] key_name The Shared Access Key Name (Policy Name). This is optional. For security
+ *                     reasons we recommend using one key per device instead of using a global
+ *                     policy key.
+ * @param[in] mqtt_password An empty #az_span with sufficient capacity to hold the MQTT password.
+ * @param[out] out_mqtt_password The output #az_span containing the MQTT password.
+ * @return #az_result.
```

Comment:
nit: there are several places where `#az_result` is followed by a `.` this should probably be removed

Reply:
Agreed.  I think there should not be the period, because just \#az_result by itself is not a complete sentence.  We need consistency at least.

I see some other places like az_span.h and az_json.h where we do indeed have a `.`, but these are complete sentences.

@CIPop az_iot_hub_client.h is itself inconsistent RE whether there's a period or not after \#az_result.  I'm removing periods from the pnp header unless you strongly object.

Reply:
I can send a PR to remove from other docs as well once the new directory structure is pushed.

Reply:
Sounds good @danewalton.  I've fixed PnP's so resolving at my level.